orca/service_test.go

```diff
@@ -93,14 +94,32 @@ func (s) TestE2E_CustomBackendMetrics_OutOfBand(t *testing.T) {
 	opts := orca.ServiceOptions{MinReportingInterval: shortReportingInterval, ServerMetricsProvider: smr}
 	internal.AllowAnyMinReportingInterval.(func(*orca.ServiceOptions))(&opts)
 
-	// Register the OpenRCAService with a very short metrics reporting interval.
+	blockCh := make(chan struct{})
```

Comment:
why did you introduce a new channel? you can continue to use mutex and define a counter requests as variable